src/server/game/map/index.ts

```diff
@@ -186,6 +210,23 @@ export class Map {
     }
   }
 
+  findRoute(pathTree: {[key: string]: Coords}, dst: {[key: string]: number}, srcPosKey: number, target: Coords): Route | null {
```

Comment:
I suppose... but I think I do need them?

Reply:
Well, the logic for the second element in the returned tuple is completely separate from the first one? So maybe only return the first one and calculate the second one somewhere else?
I'm not sure that's better; just a though.

Reply:
Hm... I do need srcPosKey anyway, and pathTree and dst are technically both parts, of the path tree. It could definitely use a cleanup, but that's a bit too big to do right now I think.